src/Nox.Types/Types/Volume/Volume.cs

```diff
@@ -1,53 +1,167 @@
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public static Volume FromCubicMeters(QuantityValue value)
-        => From(value, VolumeUnit.CubicMeter);
+    public new static Volume From(QuantityValue value)
+        => From(value, new VolumeTypeOptions());
 
     /// <summary>
-    /// Creates a new instance of <see cref="Volume"/> object in cubic meters.
+    /// Creates a new instance of <see cref="Volume"/> object in specified unit.
     /// </summary>
-    /// <param name="value">The origin value to create the <see cref="Volume"/> with</param>
+    /// <param name="value">Value to create a <see cref="Volume"/>.</param>
+    /// <param name="unit">Unit to create a <see cref="Volume"/>.</param>
     /// <returns></returns>
     /// <exception cref="TypeValidationException"></exception>
-    public new static Volume From(QuantityValue value)
-        => From(value, VolumeUnit.CubicMeter);
+    public static Volume From(QuantityValue value, VolumeUnit unit)
+        => From(value, new VolumeTypeOptions { Unit = Enum.Parse<VolumeTypeUnit>(unit.Name) });
 
-    private QuantityValue? _cubicFeet;
+    /// <summary>
+    /// Creates a new instance of <see cref="Volume"/> object in specified unit.
+    /// </summary>
+    /// <param name="value">Value to create a <see cref="Volume"/>.</param>
+    /// <param name="unit">Unit to create a <see cref="Volume"/>.</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Volume From(QuantityValue value, VolumeTypeUnit unit)
+        => From(value, new VolumeTypeOptions { Unit = unit });
+
+    /// <summary>
+    /// Creates a new instance of <see cref="Volume"/> object with specified options.
+    /// </summary>
+    /// <param name="value">Value to create a <see cref="Volume"/>.</param>
+    /// <param name="options">Options to create a <see cref="Volume"/>.</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static Volume From(QuantityValue value, VolumeTypeOptions options)
+    {
+        var newObject = new Volume
+        {
+            Value = value.Round(QuantityValueDecimalPrecision),
+            Unit = options.Unit,
+            _volumeTypeOptions = options,
+        };
 
-    public QuantityValue ToCubicFeet() => _cubicFeet ??= GetMeasurementIn(VolumeUnit.CubicFoot);
+        var validationResult = newObject.Validate();
 
-    private QuantityValue? _cubicMeters;
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Validates a <see cref="Volume"/> object.
+    /// </summary>
+    /// <returns>
+    /// True if the <see cref="Volume"/> value is valid.
+    /// </returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
```

Comment:
QuvantityValue has also a Validate method that checks whether the provided value is NaN or Infinity. 
Please use Value.Validate() instead of base.Validate() and remove the NaN and Infinity errors from this method.

You can check other measurement types for reference.

Reply:
Changed to `Value.Validate()` and repeated validations removed.